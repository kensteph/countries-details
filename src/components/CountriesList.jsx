import React, { useState } from 'react';
import Country from './Country';
import Search from './Search';

const CountriesList = () => {
  const countries = [
    {
      id: 1,
      name: 'Montserrat',
      official: 'xxxx',
      flag: 'https://flagcdn.com/w320/ms.png',
    },
    {
      id: 2,
      name: 'British Virgin Islands',
      flag: 'https://flagcdn.com/w320/cd.png',
    },
    {
      id: 11,
      name: 'Belize',
      flag: 'https://flagcdn.com/w320/bz.png',
    },
    {
      id: 15,
      name: 'Chad',
      flag: 'https://flagcdn.com/w320/td.png',
    },
    {
      id: 212,
      name: 'Spain',
      flag: 'https://flagcdn.com/w320/hu.png',
    },
    {
      id: 252,
      name: 'Vietnam',
      flag: 'https://flagcdn.com/w320/vn.png',
    },
    {
      id: 223,
      name: 'Egypt',
      flag: 'https://flagcdn.com/w320/eg.png',
    },
    {
      id: 232,
      name: 'Tokelau',
      flag: 'https://flagcdn.com/w320/tk.png',
    },
  ];
  let select = 1;
  let selectCount = 0;
  let bg = '';
  const [searchCountry, setSearchCountry] = useState('');
  let result = countries;

  const handleSearch = (ev) => {
    const searchValue = ev.target.value.toLowerCase();
    setSearchCountry(searchValue);
  };
  result = countries.filter((country) => country.name.toLowerCase().includes(searchCountry));
  return (
    <div>
      <Search handleSearch={handleSearch} />
      <div className="listContainer">
        {result.map((country, index) => {
          if (index === select) {
            bg = 'bg';
            selectCount += 1;
            // Next select
            if (selectCount === 2) {
              select = index + 3;
              selectCount = 0;
            } else {
              select = index + 1;
            }
          } else {
            bg = '';
          }
          return <Country key={country.id} className={bg} country={country} />;
        })}
      </div>
    </div>
  );
};

export default CountriesList;
