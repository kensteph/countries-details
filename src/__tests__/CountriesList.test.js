import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import CountriesList from '../components/CountriesList';

it('Should display 3 countries on the screen', () => {
  const mockStore = configureStore([]);
  const countries = [
    {
      name: 'South Africa',
      official: 'Republic of South Africa',
      currency: 'South African rand (R)',
      capital: [
        'Pretoria',
        'Bloemfontein',
        'Cape Town',
      ],
      region: 'Africa',
      subregion: 'Southern Africa',
      continents: [
        'Africa',
      ],
      languages: 'Afrikaans,English,Southern Ndebele,Northern Sotho,Southern Sotho,Swazi,Tswana,Tsonga,Venda,Xhosa,Zulu',
      latlong: '-29 , 24',
      area: 1221037,
      population: 59308690,
      flag: 'https://flagcdn.com/w320/za.png',
      views: 0,
      tld: '.za',
    },
    {
      name: 'Nauru',
      official: 'Republic of Nauru',
      currency: 'Australian dollar ($)',
      capital: [
        'Yaren',
      ],
      region: 'Oceania',
      subregion: 'Micronesia',
      continents: [
        'Oceania',
      ],
      languages: 'English,Nauru',
      latlong: '-0.53333333 , 166.91666666',
      area: 21,
      population: 10834,
      flag: 'https://flagcdn.com/w320/nr.png',
      views: 0,
      tld: '.nr',
    },
    {
      name: 'Bermuda',
      official: 'Bermuda',
      currency: 'Bermudian dollar ($)',
      capital: [
        'Hamilton',
      ],
      region: 'Americas',
      subregion: 'North America',
      continents: [
        'North America',
      ],
      languages: 'English',
      latlong: '32.33333333 , -64.75',
      area: 54,
      population: 63903,
      flag: 'https://flagcdn.com/w320/bm.png',
      views: 0,
      tld: '.bm',
    },
  ];
  const initialState = {
    countries: {
      countries,
    },
  };

  const store = mockStore(initialState);
  render(
    <BrowserRouter>
      <Provider store={store}>
        <CountriesList />
      </Provider>
    </BrowserRouter>,
  );
  const allLinks = screen.getAllByRole('link');

  expect(allLinks).toHaveLength(3);
  expect(store.getState()).toMatchSnapshot();
});
