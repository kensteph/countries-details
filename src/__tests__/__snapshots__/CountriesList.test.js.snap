// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display 3 countries on the screen 1`] = `
Object {
  "countries": Object {
    "countries": Array [
      Object {
        "area": 1221037,
        "capital": Array [
          "Pretoria",
          "Bloemfontein",
          "Cape Town",
        ],
        "continents": Array [
          "Africa",
        ],
        "currency": "South African rand (R)",
        "flag": "https://flagcdn.com/w320/za.png",
        "languages": "Afrikaans,English,Southern Ndebele,Northern Sotho,Southern Sotho,Swazi,Tswana,Tsonga,Venda,Xhosa,Zulu",
        "latlong": "-29 , 24",
        "name": "South Africa",
        "official": "Republic of South Africa",
        "population": 59308690,
        "region": "Africa",
        "subregion": "Southern Africa",
        "tld": ".za",
        "views": 0,
      },
      Object {
        "area": 21,
        "capital": Array [
          "Yaren",
        ],
        "continents": Array [
          "Oceania",
        ],
        "currency": "Australian dollar ($)",
        "flag": "https://flagcdn.com/w320/nr.png",
        "languages": "English,Nauru",
        "latlong": "-0.53333333 , 166.91666666",
        "name": "Nauru",
        "official": "Republic of Nauru",
        "population": 10834,
        "region": "Oceania",
        "subregion": "Micronesia",
        "tld": ".nr",
        "views": 0,
      },
      Object {
        "area": 54,
        "capital": Array [
          "Hamilton",
        ],
        "continents": Array [
          "North America",
        ],
        "currency": "Bermudian dollar ($)",
        "flag": "https://flagcdn.com/w320/bm.png",
        "languages": "English",
        "latlong": "32.33333333 , -64.75",
        "name": "Bermuda",
        "official": "Bermuda",
        "population": 63903,
        "region": "Americas",
        "subregion": "North America",
        "tld": ".bm",
        "views": 0,
      },
    ],
  },
}
`;
